### PR TITLE
Moj sso oauth2

### PR DESCRIPTION
**Pull request by tatyree**
This has been run and checked against a local version of the MoJ SSO. I am anticipating additional work will be required when it comes time to deploy it against the real server. It may also be the case that the workflow needs refining

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -1,3 +1,10 @@
-ACCESS_KEY_ID=
-SECRET_ACCESS_KEY=
-BUCKET_NAME=
+ACCESS_KEY_ID=dummy key
```
Comment (digitalronin) [inline .env.example]: I'd really like to have some comments in here, detailing what the significance is of these vars, and where their values come from

Comment (digitalronin) [inline .env.example]: Sorry - that'll teach me to comment before reading further down! Ignore me.

Comment (tatyree) [inline .env.example]: See the README :)

Comment (tatyree) [inline .env.example]: However, happy to put them in here as well if you'd like.

File: lib/tax_tribunal/login.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      end
+    end
+
+    get '/logout' do
+      session.destroy
+      erb :root
+    end
+
+    get '/oauth/callback' do
+      resp = authorise!(params[:code])
```
Comment (digitalronin) [inline lib/tax_tribunal/login.rb]: You could combine these two lines as `if resp = [whatever]` but that's a matter of taste, I guess

Comment (tatyree) [inline lib/tax_tribunal/login.rb]: I did.  Rubocop didn't like it, so I broke them out.

**Pull request merged**